providers/base/bin/wwan_tests.py

```diff
@@ -383,21 +412,29 @@ def invoked(self):
         args = self.register_argument()
         ret_code = 1
         try:
-            mm = MMCLI()
-            mm_id = mm.equipment_id_to_mm_id(args.hw_id)
-            _wwan_radio_on()
-            cmd = [
-                "mmcli",
-                "-m",
-                str(mm_id),
-                "--3gpp-scan",
-                "--timeout",
-                str(args.timeout),
-            ]
-            ret_code = subprocess.check_call(cmd)
+            with WWANTestCtx(args.hw_id, True, True) as ctx:
+                cmd = [
+                    "mmcli",
+                    "-m",
+                    str(ctx.modem_idx),
+                    "--3gpp-scan",
+                    "--timeout",
+                    str(args.timeout),
+                ]
+                print_head("Scanning 3GPP available network")
+                print("+ {}".format(" ".join(cmd)))
+                ret_sp = subprocess.run(
+                    cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT
+                )
+                print(ret_sp.stdout.decode("utf-8"))
+                if (
+                    ret_sp.returncode == 0
+                    and b"error" not in ret_sp.stdout.lower()
```

Comment:
Why is this needed? Are there cases where the return code from the 3gpp scanning is `0` but there is still an error?

Reply:
just in case that the `3gpp-scan` will always return `0` no matter the connection is detected or not.

BTW, here is the output while the system can't scan any available connection. 
```
admin@C0310A2043-00843:~$ sudo mmcli -m 0 --3gpp-scan --timeout 200; echo $?
error: couldn't scan networks in the modem: 'Timeout was reached'
1
```

Reply:
Thanks @pieqq  for the suggestion, I have sent a fixed.